pkg/app/base_app.go

```diff
@@ -34,16 +37,47 @@ type baseApp struct {
 
 	// LibUpdater updates ksonnet lib versions.
 	libUpdater KSLibUpdater
+	// libPath caches ksonnet lib paths after generation / validation
+	libPaths map[string]string
 
 	config    *Spec
 	overrides *Override
 
 	mu sync.Mutex
 
-	load func() error
+	load   func() error
+	loaded bool
 }
 
-func newBaseApp(fs afero.Fs, root string, httpClient *http.Client) *baseApp {
+var _ App = (*baseApp)(nil)
+
+// Opt is a constructor option for baseApp
+type Opt func(*baseApp)
+
+// OptLibUpdater returns an option for setting a KSLibUpdater on an App010
+func OptLibUpdater(libUpdater KSLibUpdater) Opt {
+	return func(a *baseApp) {
+		a.libUpdater = libUpdater
+	}
+}
+
+// optLoadFn overrides baseApp's load function, useful when testing.
+func optLoadFn(loadFn func() error) Opt {
+	return func(a *baseApp) {
+		a.load = loadFn
+	}
+}
+
+// optNopLoader overrides baseApp's loader to do nothing. (NOOP)
+func optNoopLoader() Opt {
```

Comment:
why have this at all instead of just using the above `optLoadFn(func() error {return nil})` ... alternately  now that optLoadFn isn't called anywhere we could just delete that?

i guess really my assertion would be we only need one of `optLoadFn` or `optNoopLoader`.